include/onnxruntime/core/framework/float16.h

```diff
@@ -41,11 +41,10 @@ inline bool operator<(const MLFloat16& left, const MLFloat16& right) { return le
 // BFloat16
 struct BFloat16 {
   uint16_t val{0};
-#if defined(USE_ROCM)
-  ORT_HOST_DEVICE BFloat16() = default;
-#else
-  BFloat16() = default;
+#if defined(__clang__) && defined(__HIP__)
+  ORT_HOST_DEVICE
```

Comment:
Style, nit: Although this avoids duplication, I think the older style improves readability, i.e.,

```
#if defined (__clang__) && defined(__HIP__)
  ORT_HOST_DEVICE BFloat16() = default;
#else
  BFloat16() = default;
#endif
```